src/cli/commands/publish.js

```diff
@@ -127,6 +127,27 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     throw new MessageError(reporter.lang('unknownFolderOrTarball'));
   }
 
+  // save current version so we can restore it in case of an abort
+  const oldVersion = await getVersion(config);
+
+  // sorry if this try/catch breaks your stack trace :'(
+  try {
+    return await _run(config, reporter, flags, args, pkg, dir);
+  } catch(e) {
+    try {
+      // try to restore the version
+      // note that if it crashed on setting the version, the original
+      // version probably wasn't changed in the first place
+      reporter.step(1, 0, reporter.lang('restoringVersion'));
+      await setVersion(config, reporter, {newVersion: oldVersion}, [], false);
+    } catch (_) {
+      // this was a last ditch effort to restore the version. ignore errors here
+    }
+
+    throw new Error(e);
```

Comment:
You can just rethrow `e` :)

Reply:
Cause of habit, but you're right. In this case we don't want to change the stack trace at all.